### PR TITLE
Tabzilla pr

### PR DESCRIPTION
**Pull request by penfever**
Addition of code which allows tabzilla models to run on a subset of data for a given experiment

### CODE REVIEW

File: TabZilla/tabzilla_utils.py [hunk 1]
```diff
@@ -206,10 +207,9 @@ def cross_validation(model: BaseModel, dataset: TabularDataset, time_limit: int,
     }
 
     start_time = time.time()
-
+    print("Current model: ", model)
```
Comment (duncanmcelfresh) [inline TabZilla/tabzilla_utils.py:210]: if this is for debugging only, please remove

File: TabZilla/tabzilla_utils.py [hunk 2]
```diff
@@ -226,40 +226,42 @@ def cross_validation(model: BaseModel, dataset: TabularDataset, time_limit: int,
...
         timers["train"].start()
         # loss history can be saved if needed
+        print("Fitting model, iteration ", i, " of ", len(dataset.split_indeces))
         loss_history, val_loss_history = curr_model.fit(
             X_train,
             y_train,
             X_val,
             y_val,
         )
+        print("Done fitting model, history is: ", loss_history, val_loss_history)
```
Comment (duncanmcelfresh) [inline TabZilla/tabzilla_utils.py:247]: this could produce a lot of output, I would rather not include this print statement

File: TabZilla/tabzilla_utils.py [hunk 3]
```diff
@@ -287,6 +289,7 @@ def cross_validation(model: BaseModel, dataset: TabularDataset, time_limit: int,
         predictions["test"].append(test_predictions.tolist())
         probabilities["test"].append(test_probs.tolist())
         ground_truth["test"].append(y_test.tolist())
+        print("Sample accuracy scores from test set splits: ", scorers["test"].accs)
```
Comment (duncanmcelfresh) [inline TabZilla/tabzilla_utils.py:292]: if this is for debugging alone, please remove

**Review by duncanmcelfresh — CHANGES_REQUESTED** [review]
some minor comments, mostly unneeded code. please address these and then I can merge

File: TabZilla/tabzilla_data_processing.py [hunk 4]
```diff
@@ -1,9 +1,101 @@
...
+        Make a subset of the data matrix X, with subset_features features and subset_rows rows.
+        :param X: data matrix
+        :param y: labels
+        :param subset_features: number of features to keep
+        :param subset_rows: number of rows to keep
+        :param subset_features_method: method to use for selecting features
+        :param subset_rows_method: method to use for selecting rows
+        :return: subset of X, y
+        """
+        np.random.seed(seed)
```
Comment (penfever) [inline TabZilla/tabzilla_data_processing.py:77]: Setting random seed, passed in as an argument

File: TabZilla/tabzilla_featurizer.py [hunk 5]
```diff
@@ -62,7 +62,9 @@ def featurize_dataset(dataset_path):
             [0],
             scaler="None",
             one_hot_encode=False,
-            impute=True, )
+            impute=True,
+            args=None,
```
Comment (penfever) [inline TabZilla/tabzilla_featurizer.py:66]: Modifying to match signature

**Review by duncanmcelfresh — APPROVED** [review]
